amethyst_rendy/src/batch.rs

```diff
@@ -298,6 +298,10 @@ where
         self.data_list.extend(data);
         let added_len = self.data_list.len() as u32 - start;
 
+        if added_len == 0 {
```

Comment:
Note that the `data` is an iterator, so i can't do that. I'm instead diffing lengths after extend to work that around. I was previously using iterators with known size, but it's actually quite limiting.

Also this really shoudn't happen in practice, but prevent bugs. I have considered assert too, but I think it would be a little bit too harsh, there are legitimate uses where having to handle zero outside would be just annoying.

Reply:
Ah right, I understand. I mean theoretically you could do it by using a `peekable` iterator and the peek whether or not the next item is some, but the added overhead for that would probably outweigh the benefits.